packages/backend/src/cache/cache.ts

```diff
@@ -0,0 +1,46 @@
+import { CacheDataStore, CacheSupport } from './index'
+
+export interface Cached {
+  expiry: number
+  data: CacheSupport
+}
+
+export function createInMemoryDataStore(keyTtlMs: number): CacheDataStore {
```

Comment:
We have an existing `createInMemoryDataStore` in `rafiki/packages/backend/src/middleware/cache/data-stores/in-memory.ts`. The difference between that and this seems to what is being cached. Strings vs. `CacheSupport` which is `Asset | WalletAddress | etc.`. Perhaps we can do better than a new `createInMemoryStore` for `CacheSupport` and instead parameterize the type using generics on the existing `createInMemoryStore`. In other words, a function signature like `function createInMemoryDataStore<T>(keyTtlMs: number): CacheDataStore<T>`.

Gets rid of this extra factory and makes the interfaces for the Asset, etc. strongly typed. Whereas currently we're asserting that its an `Asset` etc. on read.

For example, when using this we're doing const `inMem = (await deps.cacheDataStore.get(id)) as Asset`. If we instead initialize a cache per resource like:

```ts
// backend/src/app.ts
    return await createAssetService({
      logger: logger,
      knex: knex,
      accountingService: await deps.use('accountingService'),
      cacheDataStore: createInMemoryDataStore<Asset>(10000)
    })
```

Then instead of `get` returning `Promise<CacheSupport>` it would return `Promise<Asset>`

Reply:
Resolved.